File: students/students/src/MON.SPA/src/models/diploma/basicDocumentSubjectModel.js
export class BasicDocumentSubjectModel {
  constructor(obj = {}) {
    this.uid = obj.uid || Math.floor((Math.random()) * 0x10000).toString(16);
    this.id = obj.id;
    this.subjectCanChange = obj.subjectCanChange;
    this.position = obj.position || 1;
    this.subjectId = obj.subjectId;
    this.subjectTypeId = obj.subjectTypeId;
    this.subjectTypeDropdown = obj.subjectTypeDropdown || null;
    this.subjectDropdown = obj.subjectDropdown || null;
  }
}

File: students/students/src/MON.Models/BasicDocumentSubjectModel.cs
namespace MON.Models
{

    public class BasicDocumentSubjectModel
    {
        public int? Id { get; set; }
        public int Position { get; set; }
        public int? ParentId { get; set; }
        public bool SubjectCanChange { get; set; }
        public DropdownViewModel SubjectDropDown { get; set; }
        public int? SubjectId { get; set; }
        public int? SubjectTypeId { get; set; }
        public string Uid { get; set; }
    }
}
